### PR TITLE
builtin: add instr built-in function

### PR DESCRIPTION
**Pull request by spongedu**
Add implementation for built-in function: instr(str, substr)

### CODE REVIEW

**Comment by zimulala** [comment]
@spongedu 
Thanks for your PR, But this PR  is duplicated with the [PR](https://github.com/pingcap/tidb/pull/2857).
Can you try to implement another one?

**Comment by spongedu** [comment]
@zimulala [pr](https://github.com/pingcap/tidb/pull/2857) gives wrong result when parameter 'str' contains Chinese characters.  You may take mine or remind the author of this.

**Comment by XuHuaiyu** [comment]
@spongedu If possible, please check and register the builtin function you determine to implement in [issue](https://github.com/pingcap/tidb/issues/2863) to let others know, thank you!

File: expression/builtin_string.go [hunk 1]
```diff
@@ -1598,7 +1598,39 @@ type builtinInstrSig struct {
...
+	s, err := args[0].ToString()
+	if err != nil {
+		return d, errors.Errorf("Instr invalid str args, need string but get %T", args[0].GetValue())
+	}
+	p, err := args[1].ToString()
+	if err != nil {
+		return d, errors.Errorf("Instr invalid substr args, need string but get %T", args[0].GetValue())
+	}
+	pos := strings.Index(s, p)
+	// Not found
```
Comment (hanfei1991) [inline expression/builtin_string.go]: add `.`

File: expression/builtin_string.go [hunk 2]
```diff
@@ -1598,7 +1598,39 @@ type builtinInstrSig struct {
...
+	if pos == -1 {
+		d.SetInt64(0)
+		return d, nil
+	}
+	c := 1
+	for i := 0; i < pos; i++ {
+		if (s[i] & 0xc0) != 0x80 {
```
Comment (hanfei1991) [inline expression/builtin_string.go]: add comment for this line.

**Comment by spongedu** [comment]
@XuHuaiyu  ok. thx

File: expression/builtin_string.go (hunk 1 above)
Comment (spongedu) [inline expression/builtin_string.go]: @hanfei1991 don't get your point......

File: expression/builtin_string.go (hunk 2 above)
Comment (spongedu) [inline expression/builtin_string.go]: ok

File: expression/builtin_string.go (hunk 1 above)
Comment (hanfei1991) [inline expression/builtin_string.go]: complete your comment and add `.` at end of the sentence......

Comment (spongedu) [inline expression/builtin_string.go]: // ok.

File: expression/builtin_string.go [hunk 3]
```diff
@@ -1618,14 +1618,14 @@ func (b *builtinInstrSig) eval(row []types.Datum) (d types.Datum, err error) {
...
-	// Not found
+	// Not found.
 	if pos == -1 {
 		d.SetInt64(0)
 		return d, nil
 	}
 	c := 1
 	// Counting utf8 characters.
-	// See: http://stackoverflow.com/questions/3911536/utf-8-unicode-whats-with-0xc0-and-0x80
+	// See: 'http://stackoverflow.com/questions/3911536/utf-8-unicode-whats-with-0xc0-and-0x80' for detais.
```
Comment (hanfei1991) [inline expression/builtin_string.go]: detai->detail .............

Comment (spongedu) [inline expression/builtin_string.go]:
~~~

**Comment by zimulala** [comment]
@spongedu Yep, but we initially review in the order of the PR.

**Comment by spongedu** [comment]
@zimulala OK.

**Comment by hanfei1991** [comment]
LGTM

**Comment by spongedu** [comment]
Duplicate with this [PR](https://github.com/pingcap/tidb/pull/2857).

**Pull request closed without merging**